scripts/run-pre-build-checks

```diff
@@ -1,43 +1,57 @@
 #!/usr/bin/env bash
 
+set -e -o pipefail -u
+
 # We don't want `less` to open for `git grep` results in the below scripts when running locally.
 export GIT_PAGER=cat
 
-PATH=./scripts:$PATH
-
-# `-x`, so that we have more clarity which check actually failed
-# (rather than searching the right script by error message).
-set -e -x
-
-enforce-change-notes-updated-on-pr-to-master
-enforce-astub-explicit-annotation-import
-enforce-indent-two-spaces-outside-java
-enforce-issue-number-for-todos
-enforce-newline-at-eof
-enforce-properties-keys-with-valid-class-names
-enforce-settings-gradle-match-directories
-enforce-shell-scripts-pass-shellcheck
-enforce-version-bump
-
-if [[ $0 != *.git/hooks/pre-commit ]]; then
-  # This one is needed as a pull request check. So, it should not run as a git pre-commit hook.
-  prohibit-direct-merges-to-master
-fi
-prohibit-java-class-name-hardcode-in-strings
-prohibit-java-code-commented-out
-prohibit-java-lines-split-needlessly
-prohibit-java-unused-custom-log
-prohibit-kotlin-undefined-property-keys
-prohibit-kotlin-unsafe-cast
-prohibit-non-ascii-characters
-prohibit-properties-unused-keys
-prohibit-tab-character
-prohibit-trailing-whitespace
-
-validate-icons
-if [[ $0 != *.git/hooks/pre-commit ]]; then
-  # This one needs to connect to the linked websites and can potentially run long,
-  # let's skip it when this script is executed as git pre-commit hook.
-  validate-markdown-links
+# Note that we could easily rely on `set -e`, but then we'd have a fail-fast behavior,
+# while we want to display all failures collectively (and still fail the entire suite if any individual script failed).
+failed=""
+
+function _() {
+  script=$1
+  option=${2-}
+  if [[ $option == "--skip-in-pre-commit-hook" ]] && [[ $0 == *.git/hooks/pre-commit ]]; then
+    echo "(skipping $script as we are running in a git pre-commit hook)" >&2
+    return
+  fi
+  echo "> $script" >&2
+  if ! eval "./scripts/$script"; then
```

Comment:
`eval` is redundant, this would also work:
```suggestion
  if ! "./scripts/$script"; then
```
Leaving it just for clarity (if anything like clarity could ever be achieved in shell scripts :trollface:)